[AERIE-1934] Add metadata to activity directives

dyst5422 opened the pull request:
* **Tickets addressed:** AERIE-1934
* **Review:** By commit  <!-- Choose from: "by commit", "by file" -->
* **Merge strategy:** Merge (no squash)  <!-- Choose from: "merge (no squash)", "squash and merge" -->

## Description
<!-- What approach was taken to satisfy the ticket being addressed? What should reviewers be aware of? -->
The timestamps were pretty much copy-paste of the same from command expansion.
The name was an interesting one because we have to dynamically generate sequences for incrementation. Because of the scoping of the sequences, their names are dynamically generated as well. I have a vague concern about maintaining all the sequences over the lifetime of a mission, but I'm not sure that's founded.

## Verification
<!-- How were the changes validated? Were any automated tests added, updated, removed, or re-baselined? -->
I did a local deploy and tested out with the creation of a number of activities of different types across multiple plans.

## Documentation
<!-- What documentation was invalidated by these changes? Which artifacts should reviewers check for accuracy and completeness? -->
Documentation added as comments in SQL.

## Future work
<!-- What next steps can we anticipate from here, if any? -->

jdylanstewart committed:
Add some metadata to activity directives

goetzrrGit commented on merlin-server/sql/merlin/tables/activity.sql:
```diff
@@ -83,3 +95,41 @@ create trigger increment_revision_on_delete_activity_trigger
+returns trigger
+security definer
+language plpgsql as $$begin
+  execute 'create sequence if not exists activity_name_' || new.plan_id || '_' || new.type;
+  new.name = new.type || ' ' || nextval('activity_name_' || new.plan_id || '_' || new.type);
+  return new;
+end$$;
+
+comment on function generate_activity_name() is e''
+  'Generates a name for an activity as the activity type + number from sequence of the type scoped to its plan.';
+
+create trigger generate_name_trigger
+before insert on activity
+for each row execute function generate_activity_name();
+
+comment on trigger generate_name_trigger on activity is e''
+  'Generates a name for an activity as the activity type + number from sequence of the type scoped to its plan.';
```
More of an understanding question:

This is kinda confusing me. What is a sequence with respect to an activity?  Is it something with the Schedular?

dyst5422 replied:
Sequence as in a Postgres Sequence ie a numerically incrementing series of integers.

dyst5422 replied:
In this case, it means that for a given plan, each activity directive with the same activity type will have the name `<activityType> 1`, `<activityType> 2`, `<activityType> 3`, etc

goetzrrGit replied:
Ahhh, thank you for the clarification.

goetzrrGit reviewed (approved):
No real issues just a question.

mattdailis commented on merlin-server/sql/merlin/tables/activity.sql:
```diff
@@ -83,3 +95,41 @@ create trigger increment_revision_on_delete_activity_trigger
+create trigger set_timestamp
+  before update on activity
+  for each row
+execute procedure activity_set_updated_at();
```
Just for my own understanding - what's the difference between `execute function` and `execute procedure`, and why do we use one for `generate_activity_name`, and the other for `activity_set_updated_at`?

dyst5422 replied:
From https://www.postgresql.org/docs/current/sql-createtrigger.html

`In the syntax of CREATE TRIGGER, the keywords FUNCTION and PROCEDURE are equivalent, but the referenced function must in any case be a function, not a procedure. The use of the keyword PROCEDURE here is historical and deprecated.`

dyst5422 replied:
I went ahead and added a commit normalizing our codebase to using `execute function`

mattdailis replied:
Sounds good, thanks!

jdylanstewart committed:
Normalize use of execute function for triggers

Replaced uses of execute procedure

See https://www.postgresql.org/docs/current/sql-createtrigger.html

`In the syntax of CREATE TRIGGER, the keywords FUNCTION and PROCEDURE
are equivalent, but the referenced function must in any case be a
function, not a procedure. The use of the keyword PROCEDURE here is
historical and deprecated.`

jdylanstewart committed:
Add tags and change name generation
